### DISCUSSION TITLE
Upgrade OpenDTU to latest

### DISCUSSION THREAD

**Discussion by niekniek89** [tbnobody/OpenDTU#717] [General]
Hello,

I have a question:

last Tuesday I installed my ESP32 with OpenDTU (with the latest version that day).
I did this with the "ESP-flasher" tool (selecting only the "opendtu-generic.bin" .

now i see:

If you are upgrading from a version before 15.03.2023 you have to upgrade the partition table of the ESP32. Please follow the documentation!

do i have to flash my ESP32 from scratch now, or can i just upgrade it?

![image](https://user-images.githubusercontent.com/48858098/226161777-2dcec389-32c8-4313-8ea5-eec2f3efd080.png)

**Comment by Solarteur** [comment]
Please read the explanations here: https://github.com/tbnobody/OpenDTU/blob/master/docs/UpgradePartition.md

**Reply by niekniek89** [reply]
Thanks for your help.
I had actually seen this site before.

my question was, since I had the most recent version (Tuesday), should I run this as well.
but apparently everyone has to do this? whether you had the most recent version or not.

**Reply by jaybee79** [reply]
Hi,

you have to backup your configuration (main configuration), maybe also the pin mapping. After that the ESP should be deleted and flashed from scratch. I use the precompiled files, so I uploaded them all. After that the ESP will open a hotspot. Log in, change password (maybe not needed) and then upload the configuration file(s) again. For me this procedure looks like it's really needed because the size of the sketch seems to be at the limit with the actual partition size, so it was expanded about 30% that should be enough for the next upgrades and features.

**Reply by niekniek89** [reply]
@jaybee79 

I see that the "sketch" percentage is at 80%. how much is that for you now after a new install?

**Reply by jaybee79** [reply]
Actually I have 76% now. Before that it was nearly 100% full, so only a few kByte left.

**Comment by Solarteur** [comment]
The upgrade the partition table of the ESP32 is necessary  to extend the firmware partition. This would help for further extensions. With the new partition table, the "Memory Information" looks like this: 
![grafik](https://user-images.githubusercontent.com/117977338/226164005-fbd3fac7-a25e-4a94-b3b6-22f456b02552.png)

**Reply by niekniek89** [reply]
Hello,

this is what i'm currently seeing:

![image](https://user-images.githubusercontent.com/48858098/226164254-6da56ad9-ff83-44cc-9241-bb7bcbbed618.png)

I had already put the most recent update on it last Tuesday.
after Tuesday there was 1 new release, I see now.

**Comment by niekniek89** [comment]
@Solarteur,

so for me the partition label should already be good for future upgrades?

**Comment by Solarteur** [comment]
thnobody writes:
!! IMPORTANT UPGRADE NOTES !!

If you are upgrading from a version before 15.03.2023 you have to upgrade the partition table of the ESP32. Please follow the this documentation!
(https://github.com/tbnobody/OpenDTU#-important-upgrade-notes-)

My understanding of the sentence is that for all "clean installs" done before 03/15/2023, the partition table must also be reinstalled.

thnobody writes:
My understanding of the sentence is that for all "clean installs" done before 03/15/2023, the partition table must also be reinstalled.

If I'm wrong about that - please let me know.

**Reply by niekniek89** [reply]
thanks for your answer.
yes, I find it confusing whether or not I need to reinstall everything.

I did not upgrade before 3/15/2023, but a clean install (new OpenDTU user)

**Comment by tbnobody** [comment]
What is confusing? If you are running a version from before 15.03.2023 e.g. 14.03.2023 or 13.03.2023 or 01.10.2022 you have to do a clean reinstall. 
If you did a clean install after 15.03.2023 you are good to go.

**Reply by niekniek89** [reply]
thanks @tbnobody for your help.
i install this release:

![image](https://user-images.githubusercontent.com/48858098/226289114-ee3302e8-f5ef-46d1-9b4f-d5092a6c76d7.png)

I'll be fine with this version.

**Reply by tbnobody** [reply]
But be aware, that this is not a official version. 
The readme states:
> (search for "master" in the blue font text but click on the white header text!).

You installed "LucaDev:master". That's a Pull Request from another user with it's modifications. As written in the readme, search for "master"....

**Comment by gcPhil** [comment]
If I understand it correctly, I must be happy that my opendtu still works at all with the current version d508b41, even without a clean install upgrade. **It would be nice if such important messages appeared noticeably on the firmware update page or at least on the system status page.** With the amount of updates, it's annoying enough to have to generate a new build every few days, so I guess not everyone reads through all the commits carefully.

**Reply by tbnobody** [reply]
You don't have to read all the commits... It appears in very large letters on the Start Page of this github repo:
![grafik](https://user-images.githubusercontent.com/1753222/226571774-5ce7dc03-2bae-4330-b47c-c1591d7a2406.png)

Breaking changes are also shown in the Readme.
![grafik](https://user-images.githubusercontent.com/1753222/226572074-aafd39bf-427e-42d4-8f09-53a8b9869f70.png)


If you are so annoyed just don't update or use the original DTU.

**Reply by gcPhil** [reply]
Sorry, that may have been misunderstood. Of course I'm pleased that the project is developing so quickly and I'm curious about every new update. But as someone who is not an expert in electronics or programming languages, and who is not a native English speaker, it is very difficult to understand all this.
**I was just glad I didn't brick the ESP because I didn't compare the readme file or visit the start page. My comment was not meant to express dissatisfaction**, but just to point out that I'm sure many people feel the same way I do. It would be really helpful for us if there was a more obvious indication in the backend as soon as an update could not be safely installed as usual.

**Reply by tbnobody** [reply]
It is not possible to brick an ESP. The worst thing would could happen is that you have to reflash it again via usb.

**Comment by cbecker2** [comment]
Hi folks,
back in February i've flashed my ESP32 with the tool ESP-Flasher provided with this link (https://github.com/Jason2866/ESP_Flasher/releases/) on your github page. 
Since i'm totally new in flashing these devices, i also need to know whether it's enough to simply "re-flash" a new opendtu-generic.bin (and not doing an OTA update) using this ESP-Flasher to achieve the new partition layout?
Sorry if i ask the same question again....
Regards
Christian

**Comment by Solarteur** [comment]
It's fine to simply "re-flash" a new opendtu-generic.bin using the ESP flasher linked above to achieve the new partition layout.

"OTA" is then possible again for future updates.

**Reply by cbecker2** [reply]
Thx - that's THE answer for me!

**Comment by niekniek89** [comment]
thanks for everyone's help!
i just upgraded OTA to the latest release.
Sketch percentage dropped from 80% to 78%.

thanks for developing this beautiful tool!

**Comment by ele-dev** [comment]
I have a question related to the whole reflashing thing due to changes in the partition table.
I know I can backup pin_mapping and main configuration but what's included in those files.
Is it possible the restore the exact state of the system including the totalYield Value or is the loss of some data inevitable.
Is the totalYield counter stored on the inverter itself or somehow on the ESP and therefore gone after reflashing?

I ask this because I don't want to mess up my MQTT data collection and visualization tool chain.

**Reply by Solarteur** [reply]
OpenDTU is the bridge between the inverter and a WLAN. The "operating data" of the inverter is stored there and can be called up again after an OpenDTU update.
